app/pages/learn/protocols/stakewise-v2.mdx

```diff
@@ -0,0 +1,212 @@
+
+<Callout type="info">
+  In the Uniswap v3 PositionsNFT contract, the `decreaseLiquidity` function has
+  a special modifier called `isAuthorizedForToken`. This modifier acts like a
+  gatekeeper, ensuring that only authorized senders can perform this action.
+  Uniswap v3 has these checks built into its system, so there's no need for
+  additional constraints.
+</Callout>
+
+### `collect · PositionsNFT`
+
+Collect accumulated fees from a specific NFT position.
+
+- **Contract:** `PositionsNFT`
+- **Function:** `collect((uint256,address,uint128,uint128))`
+
+  | Parameter           | Constraint        |
+  | :------------------ | :---------------- |
+  | `params.tokenId`    | -                 |
+  | `params.recipient`  | equal to `AVATAR` |
+  | `params.amount0Max` | -                 |
+  | `params.amount1Max` | -                 |
+
+<Callout type="info">
+  In the Uniswap v3 PositionsNFT contract, the `collect` function has a special
+  modifier called `isAuthorizedForToken`. This modifier acts like a gatekeeper,
+  ensuring that only authorized senders can perform this action. Uniswap v3 has
+  these checks built into its system, so there's no need for additional
+  constraints.
+</Callout>
+
+### `MerkleDistributor · claim`
+
+Claim accrued rewards in `rETH2` and `SWISE` tokens.
+
+- **Contract:** `MerkleDistributor`
+- **Function:** `	claim(uint256,address,address[],uint256[],bytes32[])`
+
+  | Parameter     | Constraint                             |
+  | :------------ | :------------------------------------- |
+  | `index`       | -                                      |
+  | `account`     | equal to `AVATAR`                      |
+  | `tokens`      | equal to a subset of (`rETH2`,`SWISE`) |
+  | `amounts`     | -                                      |
+  | `merkleProof` | -                                      |
+
+### If WETH is involved, the following functions are also whitelisted:
+
+#### `collect · PositionsNFT`
+
+Collect accumulated fees from a specific NFT position.
+
+- **Contract:** `PositionsNFT`
+- **Function:** `collect((uint256,address,uint128,uint128))`
+
+  | Parameter           | Constraint                                            |
+  | :------------------ | :---------------------------------------------------- |
+  | `params.tokenId`    | -                                                     |
+  | `params.recipient`  | equal to `0x0000000000000000000000000000000000000000` |
+  | `params.amount0Max` | -                                                     |
+  | `params.amount1Max` | -                                                     |
+
+<Callout type="info">
+  In the Uniswap v3 PositionsNFT contract, the `collect` function has a special
+  modifier called `isAuthorizedForToken`. This modifier acts like a gatekeeper,
+  ensuring that only authorized senders can perform this action. Uniswap v3 has
+  these checks built into its system, so there's no need for additional
+  constraints.
+</Callout>
+
```

Comment:
It's ok to mention this again, or should we remove it?

Reply:
Maybe we should also add this warning to stakewise:
<Callout type="warning">
  If the avatar has approved the Uniswap v3 Positions NFT contract to spend
  tokens other than the ones specified in `tokens`, the role will be able to
  increase any existing positions' liquidity in these tokens as well.
</Callout>

Reply:
> It's ok to mention this again, or should we remove it? Should we add these comments in univ3?

I'd say it's not really necessary to mention it. Should be obvious.. If Uniswap would not enforce these constraints anyone could exploit this.

Reply:
adding the warning about open approvals makes total sense